cmd/fs-v1-metadata.go

```diff
@@ -61,21 +64,33 @@ type fsMetaV1 struct {
 	Parts []objectPartInfo  `json:"parts,omitempty"`
 }
 
+// A fsMetaV2 represents a metadata header mapping keys to sets of values.
+type fsMetaV2 struct {
+	Version  string `json:"version"`
+	Checksum struct {
+		Algorithm string `json:"algorithm"`
+		BlockSize string `json:"blocksize"`
+		Hash      string `json:"hash"`
+	} `json:"checksum,omitempty"`
+	// Metadata map for current object `fs.json`.
```

Comment:
`Metadata map for current object `fs.json``. this comment can be a bit better at the end having `fs.json` doesn't make sense.